code/modules/food_and_drinks/machinery/coffeemaker.dm

```diff
@@ -186,7 +200,66 @@
 		if(!user.transferItemToLoc(new_pot, src))
 			return TRUE
 		replace_pot(user, new_pot)
-		balloon_alert(user, "added pot")
+		update_appearance()
+		return TRUE //no afterattack
+
+	if (istype(attack_item, /obj/item/reagent_containers/cup/glass/coffee_cup) && !(attack_item.item_flags & ABSTRACT) && attack_item.is_open_container())
+		var/obj/item/reagent_containers/cup/glass/coffee_cup/new_cup = attack_item
+		if(new_cup.reagents.total_volume > 0 )
+			balloon_alert(user, "the cup must be empty!")
+			return TRUE
+		if(coffee_cups >= max_coffee_cups)
+			balloon_alert(user, "the cup holder is full!")
+			return TRUE
+		. = TRUE //no afterattack
+		if(!user.transferItemToLoc(attack_item, src))
+			return TRUE
+		coffee_cups++
+		update_appearance()
+		return TRUE //no afterattack
+
+	if (istype(attack_item, /obj/item/reagent_containers/condiment/pack/sugar))
+		var/obj/item/reagent_containers/condiment/pack/sugar/new_pack = attack_item
+		if(new_pack.reagents.total_volume < new_pack.reagents.maximum_volume)
+			balloon_alert(user, "the pack must be full!")
+			return TRUE
+		if(sugar_packs >= max_sugar_packs)
+			balloon_alert(user, "the sugar compartment is full!")
+			return TRUE
+		. = TRUE //no afterattack
```

Comment:
This isn't necessary.

Reply:
Do you mean as above, that I shouldn't be returning `TRUE` everywhere or that I shouldn't check for the sugar packets at all?